### ISSUE TITLE
Keyhotee loading duplicate msgs (probably last_time_sync related error)

### ISSUE THREAD

**Issue by dnotestein** [InvictusInnovations/keyhotee#238]
Opening Keyhotee again, the Inbox shows several duplicates of messages that I received during my last Keyhotee session approximately one hour ago, all with the same received time as the original messages.  I estimate 5-6 duplicate messages.

I closed and re-opened Keyhotee, and now I counted 52 (!) duplicate messages, again all with the same received time as the original messages.

I closed and re-opened Keyhotee a third time, and now I count approximately 139 duplicate messages with the same received time.

**Comment by dnotestein** [comment]
Taken from Issue #290:
Steps to reproduce:
1) open a client.
2) send 3 messages to itself one by one.
3) close the client
4) start the client again.
5) check inbox

The last message sent is twice received in inbox

Tested in Ubuntu and is always reproducible

**Comment by dnotestein** [comment]
Yuvaraj, are you able to reproduce this error when you are only connected to the mail server? I couldn't reproduce it on windows following your steps, but I have ALPHA_RELEASE enabled, so maybe we have a different testing setup.

**Comment by yuvarajgogoi** [comment]
It is not reproducible this issue now.

**Comment by dnotestein** [comment]
I changed mail server to set message timestamp when it receives a message (previously it used the time mail was sent from originator, which could have a skewed clock time). The receiver of the message sets its last_sync_time from the timestamp of the message, so a receiver could keep setting it's last_sync_time into the past when receiving a message from a sender with a clock in the past, causing continual duplicates until a later message was received which pushed the last_sync_time into the future. Assuming this is the only way this error could occur, then I fixed the server code in https://github.com/InvictusInnovations/BitShares/commit/ac66e30d912b48dbb34dd1d6b1c1b59cd4b0710a.
